core/src/services/pcloud/error.rs

```diff
@@ -0,0 +1,119 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use http::Response;
+use serde::Deserialize;
+use serde_json::Value;
+
+use crate::raw::*;
+use crate::Error;
+use crate::ErrorKind;
+use crate::Result;
+
+/// PcloudError is the error returned by Pcloud service.
+#[derive(Default, Debug, Deserialize)]
+#[allow(unused)]
+struct PcloudError {
+    result: u32,
+    error: String,
+}
+
+pub fn parse_resp(resp: &Value) -> Result<u64> {
```

Comment:
It's a bit odd to parse `serde_json::Value` directly. Can we parse it into a static struct instead?